### PR TITLE
Rounding validation

### PR DESCRIPTION
**Pull request by LogvinovLeon**
This PR:
* Adds a validations that the order values are within the range without rounding errors

### CODE REVIEW

File: packages/0x.js/src/utils/order_state_utils.ts [hunk 1]
```diff
@@ -113,6 +117,13 @@ export class OrderStateUtils {
                 throw new Error(ExchangeContractErrs.InsufficientMakerFeeAllowance);
             }
         }
+        const minimumFillableTakerTokenAmountWithingNoRoundingErrorRange = signedOrder.takerTokenAmount
```
Comment (fabioberger) [inline packages/0x.js/src/utils/order_state_utils.ts]: `minFillableTakerTokenAmountWithinNoRoundingErrorRange`

File: packages/0x.js/src/utils/order_state_utils.ts [hunk 2]
```diff
@@ -113,6 +117,13 @@ export class OrderStateUtils {
...
+                                                                           .times(1000)
```
Comment (fabioberger) [inline packages/0x.js/src/utils/order_state_utils.ts]: Why 1000? Can you set to constant with a descriptive name?

File: packages/0x.js/src/utils/order_state_utils.ts [hunk 3]
```diff
@@ -113,6 +117,13 @@ export class OrderStateUtils {
...
+                                                                           .dividedBy(signedOrder.makerTokenAmount);
+        if (orderRelevantState.remainingFillableTakerTokenAmount
+            .lessThan(minimumFillableTakerTokenAmountWithingNoRoundingErrorRange)) {
+            throw new Error(ExchangeContractErrs.OrderFillRoundingError);
```
Comment (fabioberger) [inline packages/0x.js/src/utils/order_state_utils.ts]: I think we need a more specific error message name here. Perhaps: `CannotFillOrderWithoutCausingRoundingError`?

Comment (LogvinovLeon) [reply packages/0x.js/src/utils/order_state_utils.ts]: Should it include some info about intervals?
Cause the above is not true. You can.

**Pull request merged**